fakestorage/json_response.go

```diff
@@ -53,3 +55,12 @@ func (r *jsonResponse) getErrorMessage(status int) string {
 	}
 	return http.StatusText(status)
 }
+
+func errToJsonResponse(err error) jsonResponse {
+	// TODO: map more errno
+	status := 0
+	if e, ok := err.(*os.PathError); ok && e.Err == syscall.ENAMETOOLONG {
```

Comment:
We probably want to use [errors.As](https://pkg.go.dev/errors#As), in case the error is wrapped.

Reply:
Done.